hudi-spark/src/main/scala/org/apache/hudi/IncrementalRelation.scala

```diff
@@ -65,15 +65,20 @@ class IncrementalRelation(val sqlContext: SQLContext,
       s"option ${DataSourceReadOptions.BEGIN_INSTANTTIME_OPT_KEY}")
   }
 
-  val lastInstant = commitTimeline.lastInstant().get()
+  private val lastInstant = commitTimeline.getReverseOrderedInstants.iterator().find(instant => {
+    // Skip empty instants
+    HoodieCommitMetadata
+      .fromBytes(commitTimeline.getInstantDetails(instant).get(), classOf[HoodieCommitMetadata])
+      .fetchTotalRecordsWritten() > 0
+  }).getOrElse(throw new HoodieException("Couldn't find any non empty instant in the hoodie commit timeline."))
```

Comment:
Can we return an empty relation instead of throwing an error here.

Reply:
No longer need to handle empty commit here since schema was handled by `TableSchemaResolver`